src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -125,9 +126,10 @@ internal async Task LoadPostgresInfo([NotNull] NpgsqlConnection conn, NpgsqlTime
         /// Order by primitives first, container later.
         /// For arrays and ranges, join in the element OID and type (to filter out arrays of unhandled
         /// types).
+        /// Field pg_type.typcategory is added after 8.4. refer: https://www.postgresql.org/docs/8.4/static/catalog-pg-type.html#CATALOG-TYPCATEGORY-TABLE
         /// </remarks>
         [NotNull]
-        static string GenerateTypesQuery(bool withRange, bool withEnum, bool withEnumSortOrder, bool loadTableComposites)
+        static string GenerateTypesQuery(bool withRange, bool withEnum, bool withEnumSortOrder, bool loadTableComposites, Version version)
```

Comment:
This changes is similar to the recent changes for `withEnumSortOrder`. Please take a look at #2027 and refactoring to follow that pattern.